### PR TITLE
fix entityPropsFactory id param, #21809

### PR DESCRIPTION
**Pull request by patriknw**
* it was using the shardId as the parameter to the entityPropsFactory
* some names were also wrong in the ClusterShardingSpec

Refs #21809

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 1]
```diff
@@ -800,15 +800,15 @@ private[akka] class ShardRegion(
...
+          case Some(factory) if !shardsByRef.values.exists(_ == id) ⇒
             log.debug("Starting shard [{}] in region", id)
 
             val name = URLEncoder.encode(id, "utf-8")
             val shard = context.watch(context.actorOf(
               Shard.props(
                 typeName,
                 id,
-                props(id),
+                factory,
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: this is where the Shard is created, so the id is the shardId, not the entityId

Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: Ouch

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
This fixes the non-released regression introduced in https://github.com/akka/akka/pull/24058

cc @talpr

**Comment by patriknw** [comment]
added some filters to mima, all internal stuff

**Review by 2m — APPROVED** [review]
LGTM

**Review by ktoso — APPROVED** [review]
LGTM, good catch

**Pull request merged**